Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -882,6 +861,54 @@ extension BrowserViewController {
     func hideReaderModeBar(#animated: Bool) {
         readerModeBar.hidden = true
     }
+
+    func enableReaderMode() {
+        if let webView = tabManager.selectedTab?.webView {
+            let backList = webView.backForwardList.backList as [WKBackForwardListItem]
+            let forwardList = webView.backForwardList.forwardList as [WKBackForwardListItem]
+
+            if let currentURL = webView.backForwardList.currentItem?.URL {
+                if let readerModeURL = ReaderModeUtils.encodeURL(currentURL) {
+                    if backList.count > 1 && backList.last?.URL == readerModeURL{
+                        webView.goToBackForwardListItem(backList.last!)
+                    } else if forwardList.count > 0 && forwardList.first?.URL == readerModeURL {
+                        webView.goToBackForwardListItem(forwardList.first!)
+                    } else {
+                        // Store the readability result in the cache and load it. This will later move to the ReadabilityHelper.
+                        webView.evaluateJavaScript("\(ReaderModeNamespace).readerize()", completionHandler: { (object, error) -> Void in
+                            if let readabilityResult = ReadabilityResult(object: object) {
+                                ReaderModeCache.sharedInstance.put(currentURL, readabilityResult, error: nil)
+                                webView.loadRequest(NSURLRequest(URL: readerModeURL))
+                            }
+                        })
+                    }
+                }
+            }
+        }
+    }
+
+    func disableReaderMode() {
+        if let webView = tabManager.selectedTab?.webView {
+            let backList = webView.backForwardList.backList as [WKBackForwardListItem]
+            let forwardList = webView.backForwardList.forwardList as [WKBackForwardListItem]
+
+            if let currentURL = webView.backForwardList.currentItem?.URL {
+                if let originalURL = ReaderModeUtils.decodeURL(currentURL) {
+                    if backList.count == 1 && forwardList.count == 0 {
```

Comment:
Not sure what the point of this if statement is, but you can remove it and it will have no effect (we'll just end up in the `else` block below, which does the same thing).